### PR TITLE
Windows support in host_helpers

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Add Windows support to:
  * file_exists_on
  * directory_exists_on
  * link_exists_on (fallback to file_exists_on)
  * file_contents_on
* Expand windows environment variables on scp_from and scp_to

### CODE REVIEW

File: lib/beaker/ssh_connection.rb [hunk 1]
```diff
@@ -319,6 +325,12 @@ def scp_from source, target, options = {}
       result.stdout = "\n"
 
       begin
+        # This is probably windows with an environment variable so we need to
+        # expand it.
+        source = self.execute(%{echo "#{source}"}).output.strip.gsub('"','') if source.include?('%')
+
+        require 'pry'
+        binding.pry
```
Comment (gimmyxd) [inline lib/beaker/ssh_connection.rb]: Please remove this, the CI got stuck

Comment (trevor-vaughan) [reply lib/beaker/ssh_connection.rb]: @gimmyxd Sorry about that :-|. Thought that I had caught them all.

**Comment by mihaibuzgau** [comment]
The tests that are failing seem to be unrelated to this PR.

File: lib/beaker/dsl/helpers/host_helpers.rb [hunk 2]
```diff
@@ -469,9 +469,13 @@ def backup_the_file host, current_dir, new_dir, filename = 'puppet.conf'
         #
         # @return [Boolean] Whether the file exists on the host (using `test -f`)
         def file_exists_on(host, file_path)
-          host.execute(%(test -f "#{file_path}"), accept_all_exit_codes: true) do |result|
-            return result.exit_code.zero?
+          if host['platform'] =~ /windows/
+            command = Command.new(%{IF exist "#{file_path}" exit 0 ELSE exit 1}, [], { :cmdexe => true, :accept_all_exit_codes => true })
```
Comment (trevor-vaughan) [inline lib/beaker/dsl/helpers/host_helpers.rb]: Hold on this, this isn't working for who knows what reason :-|

**Comment by trevor-vaughan** [comment]
This appears to be working as expected now

**Comment by gimmyxd** [comment]
@puppetlabs-jenkins retest this please

**Comment by gimmyxd** [comment]
I've run puppet-agent ad-hoc for master with this and https://github.com/puppetlabs/beaker-puppet/pull/131. Seems fine on our side
image

But beaker is extensively used on other project, like PE testing and maybe some modules. Can someone for @puppetlabs/installer-and-management and @puppetlabs/content @puppetlabs/modules asset impact of this on their side?

**Comment by genebean** [comment]
The failing check seems to be related to a master platform. I have pinged the team I think that needs to look at that in Slack and will follow up when they are online later today.

**Comment by genebean** [comment]
The modules team no longer uses Beaker so they have no objection to this moving forward (they have moved to Litmus).

**Comment by cthorn42** [comment]
Sorry, looks like the configuration for the testing of PE is a bit out of date. I'm going to go ahead and approve the PR and try to fix up that configuration out of band.

**Comment by trevor-vaughan** [comment]
@cthorn42 Hold on the merge please. I just found an issue with reading alternate data streams. Will have it fixed soon.

**Comment by trevor-vaughan** [comment]
@genebean and @cthorn42 This should be OK now. I had to rework some stuff in Powershell to get the ADS support working but it appears to work like a charm now.

**Comment by gimmyxd** [comment]
👍 tested again with puppet-agent ad-hoc
image

**Pull request merged**